### DISCUSSION TITLE
User feedback as metamorpho vault manager

### DISCUSSION THREAD

**Discussion by yaronvel** [morpho-org/metamorpho#379] [Ideas]
1. timelocks in initial configuration are really annoying. obvious reason is we must wait 24 hours between deployment and testing. but another issue is it prevents us from doing a full configuration in a single gnosis safe batch, as we cannot set the supply queue (or withdrawal queue) along with the market listing.
2. deploying a vault takes a lot of gas. The gas estimation by etherscan is 6m, actual consumption is around 3-4m, which is still a lot.
3. Support YOLO guardian role: normally (e.g., aave, but also some DEXes and other protocols) guardian role can be revoked by the DAO (in our case vault owner). this allows to assign guardian roles to less secure accounts that could perform quick intervention during times of emergency. In the current metamorpho design, the guardian can also prevent the owner from changing guardian role. while this is imperative to make sure the vault in non custodial, it also dictates that the guardian account/address should be very secure, as otherwise, if it gets compromised, it could block all future parameter changes. Ideally, in addition to the current guardian, there should also be a way to assign a yolo guardian (somewhat similar to allocator role), that could quickly revoke the curator operations (in case the curator got compromised), but can still be revoked by the owner. Technically this could also be done as a layer on top of existing guardian (or even curator or owner).

**Comment by Shiranm88** [comment]
> 1. timelocks in initial configuration are really annoying. obvious reason is we must wait 24 hours between deployment and testing. but another issue is it prevents us from doing a full configuration in a single gnosis safe batch, as we cannot set the supply queue (or withdrawal queue) along with the market listing.
> 2. deploying a vault takes a lot of gas. The gas estimation by etherscan is 6m, actual consumption is around 3-4m, which is still a lot.
> 3. Support YOLO guardian role: normally (e.g., aave, but also some DEXes and other protocols) guardian role can be revoked by the DAO (in our case vault owner). this allows to assign guardian roles to less secure accounts that could perform quick intervention during times of emergency. In the current metamorpho design, the guardian can also prevent the owner from changing guardian role. while this is imperative to make sure the vault in non custodial, it also dictates that the guardian account/address should be very secure, as otherwise, if it gets compromised, it could block all future parameter changes. Ideally, in addition to the current guardian, there should also be a way to assign a yolo guardian (somewhat similar to allocator role), that could quickly revoke the curator operations (in case the curator got compromised), but can still be revoked by the owner. Technically this could also be done as a layer on top of existing guardian (or even curator or owner).

**Comment by Shiranm88** [comment]
Tjo
